src/crud/useWavesurferRegions.tsx

```diff
@@ -566,7 +564,8 @@ export function useWaveSurferRegions(
     } */
   }
   function onRegionSeek(e: number, keepRegion: boolean) {
-    if (singleRegionRef.current && !keepRegion && currentRegion()) {
+    // TT-2884 remove single regions
+    if (singleRegionRef.current && currentRegion()) {
```

Comment:
This will remove the region too often.  When we are seeking into the region programmatically it needs to not remove it.  You probably need to figure out why keepRegion is true in the issue you're working on.

Reply:
The condition says singleRegionRef.current. I think this is only true when the user has created this single region. When I was testing yesterday, I used the segmentation. Then I could select into a region and I could move the region boundaries. These all worked and this code did not affect it because we weren't working on a single region.